report/sections/Chapter 07 - Queue System/Chapter 7.tex

```diff
@@ -34,7 +34,7 @@ \section{The queue}
 In our case, there are two key variables inside the shared state structure that both endpoints need access to:
 \begin{enumerate}
     \item \textbf{A sender channel:} In practice, this channel is used as a worker queue in which code submission requests are enqueued for processing. It has a corresponding \textbf{receiver channel}, which is to dequeue and schedule work. Importantly, requests sent via the sender channel are received in a first-in-first-out order.
-    \item \textbf{The job results hashmap:} Upon first receiving a code submission request, a UUID is generated and used as a key to reference an empty piece of work inside the hashmap. When the test runner has processed the code submission, it inserts the result of the work in the hashmap as the value in the hashmap using the UUID as the key. This value can then be looked up when the client requests the result processing their code submission, since the client is provided with the UUID upon sending a code submission, as previously mentioned.
+    \item \textbf{The job results hashmap:} Upon first receiving a code submission request, a UUID is generated. This is used as a key inside the job results hashmap, where the value is initially set to Rust \texttt{Option} value of type \texttt{TestRunnerResult} --- in other words, a key-value pair with the UUID referencing an empty job result, indicating that the work has not yet been completed. An \texttt{Option} type is simply a nullable data structure in Rust; an optional value. When the test runner has processed the code submission, the result is inserted as the value of the \texttt{Option} type. This value can then be looked up when the client requests the result processing their code submission, since the client is provided with the UUID upon sending a code submission, as previously mentioned.
```

Comment:
```suggestion
    \item \textbf{The job results hashmap:} Upon first receiving a code submission request, a UUID is generated. This is used as a key inside the job results hashmap, where the value is initially set to the Rust \texttt{Option} value of type \texttt{TestRunnerResult} --- in other words, a key-value pair with the UUID referencing an empty job result, indicating that the work has not yet been completed. An \texttt{Option} type is simply a nullable data structure in Rust; an optional value. When the test runner has processed the code submission, the result is inserted as the value of the \texttt{Option} type. This value can then be looked up when the client requests the result processing their code submission, since the client is provided with the UUID upon sending a code submission, as previously mentioned.
```